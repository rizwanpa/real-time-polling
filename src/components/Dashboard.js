import React, { Component } from "react";
import { connect } from "react-redux";

class Dashboard extends Component{
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div>Dashboard</div>
    )
  }
}

const mapStateToProps = state => {
  console.log("Dashboard-->", state);
  return { ...state };
};

export default connect(
  mapStateToProps
)(Dashboard);
