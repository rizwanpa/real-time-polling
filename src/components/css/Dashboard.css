.dashboard, .polls {
  width: 100%;
  height: 100%;
}
.polls {
  overflow-y: auto;
  overflow-x: hidden;
  padding: 5px 15px;
}
.poll:hover {
  background: rgba(0,0,0,0.1);
  cursor: pointer;
}
.poll {
  display: flex;
  margin-bottom: 10px;
  width: 100%;
  padding: 5px 10px;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
  -moz-box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
  -webkit-box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
  border: thin solid transparent;
  border-radius: 6px;
  user-select: none;
}
.poll-info, .poll-response {
  flex: 1;
  width: 50%;
}
.poll-desc {
  max-width: 100%;
  margin:0px;
  padding:0px;
  font-size: 80%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}
.poll-response {
  display: flex;
  justify-content: space-around;
}
.text-center {
  text-align: center;
}
.poll-question {
  display: flex;
  margin: 0 50px;
  padding: 10px;
}
.poll-question > * {
  flex: 1;
}