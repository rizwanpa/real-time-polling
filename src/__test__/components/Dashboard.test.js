import React from "react";
import { Provider } from "react-redux";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store"; // Smart components

// Component to be tested
import Dashboard from "./../../components/Dashboard";
configure({ adapter: new Adapter() });
// create any initial state needed
const initialState = {
  polls: [],
  pollsAnalytics: [],
  deletedPollId: "",
  editPoll: [],
  editDetails: {}
};
// here it is possible to pass in any middleware if needed into //configureStore
const mockStore = configureStore();
const store = mockStore(initialState);

describe("Dashboard Component", () => {
  let wrapperDashboard;
  // our mock Dashboard function to replace the one provided by mapDispatchToProps
  const mockDashboardfn = jest.fn();
  describe("render()", () => {
    test("renders the component", () => {
      const wrapper = shallow(
        <Provider store={store}>
          <Dashboard />
        </Provider>
      );
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
