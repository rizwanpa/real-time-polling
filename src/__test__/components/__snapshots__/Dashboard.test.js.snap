// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component render() renders the component 1`] = `<Connect(Dashboard) />`;
